pkg/queue/manager.go

```diff
@@ -293,25 +293,26 @@ func (m *Manager) RequeueWorkload(ctx context.Context, info *workload.Info, imme
 	m.Lock()
 	defer m.Unlock()
 
-	q := m.queues[queueKeyForWorkload(info.Obj)]
-	if q == nil {
-		return false
-	}
-
 	var w kueue.Workload
+	// Always get the newest workload to avoid requeuing the out-of-date obj.
 	err := m.client.Get(ctx, client.ObjectKeyFromObject(info.Obj), &w)
 	// Since the client is cached, the only possible error is NotFound
 	if apierrors.IsNotFound(err) || w.Spec.Admission != nil {
 		return false
 	}
 
-	q.AddIfNotPresent(info)
+	q := m.queues[queueKeyForWorkload(&w)]
+	if q == nil {
+		return false
+	}
+	q.AddOrUpdate(&w)
 	q.reportPendingWorkloads()
 	cq := m.clusterQueues[q.ClusterQueue]
 	if cq == nil {
 		return false
 	}
 
+	info = workload.NewInfo(&w)
```

Comment:
Once we do #162, we probably want to make this `info.Update(&w)` or similar.

Can you leave a `TODO(#162)`?

Reply:
Done.

Reply:
I'll keep in mind then. Thanks for pointing out.